src/SDK/Metrics/MeterProvider.php

```diff
@@ -0,0 +1,181 @@
+use OpenTelemetry\Context\ContextStorageInterface;
+use OpenTelemetry\SDK\Common\Attribute\Attributes;
+use OpenTelemetry\SDK\Common\Attribute\AttributesFactoryInterface;
+use OpenTelemetry\SDK\Common\Instrumentation\InstrumentationScopeFactoryInterface;
+use OpenTelemetry\SDK\Common\Time\ClockInterface;
+use OpenTelemetry\SDK\Metrics\AttributeProcessor\FilteredAttributeProcessor;
+use OpenTelemetry\SDK\Metrics\Exemplar\ExemplarFilter\WithSampledTraceExemplarFilter;
+use OpenTelemetry\SDK\Metrics\Exemplar\ExemplarReservoirInterface;
+use OpenTelemetry\SDK\Metrics\Exemplar\FilteredReservoir;
+use OpenTelemetry\SDK\Metrics\Exemplar\FixedSizeReservoir;
+use OpenTelemetry\SDK\Metrics\Exemplar\HistogramBucketReservoir;
+use OpenTelemetry\SDK\Metrics\MetricFactory\StreamFactory;
+use OpenTelemetry\SDK\Metrics\View\CriteriaViewRegistry;
+use OpenTelemetry\SDK\Metrics\View\FallbackViewRegistry;
+use OpenTelemetry\SDK\Metrics\View\SelectionCriteriaInterface;
+use OpenTelemetry\SDK\Metrics\View\ViewTemplate;
+use OpenTelemetry\SDK\Resource\ResourceInfo;
+
+final class MeterProvider implements MeterProviderInterface
+{
+    private ?ContextStorageInterface $contextStorage;
+    private MetricFactoryInterface $metricFactory;
+    private ResourceInfo $resource;
+    private AttributesFactoryInterface $attributesFactory;
+    private ClockInterface $clock;
+    private InstrumentationScopeFactoryInterface $instrumentationScopeFactory;
+    /** @var MetricSourceRegistryInterface&MetricReaderInterface */
+    private MetricReaderInterface $metricReader;
+    private StalenessHandlerFactoryInterface $stalenessHandlerFactory;
+    private CriteriaViewRegistry $criteriaViewRegistry;
+    private ViewRegistryInterface $viewRegistry;
+    private MeterInstruments $instruments;
+
+    private bool $closed = false;
+
+    /**
+     * @param MetricSourceRegistryInterface&MetricReaderInterface $metricReader
+     */
+    public function __construct(
+        ?ContextStorageInterface $contextStorage,
+        ResourceInfo $resource,
+        ClockInterface $clock,
+        InstrumentationScopeFactoryInterface $instrumentationScopeFactory,
+        $metricReader,
+        AttributesFactoryInterface $attributesFactory,
+        StalenessHandlerFactoryInterface $stalenessHandlerFactory,
+        MetricFactoryInterface $metricFactory = null
+    ) {
+        $this->contextStorage = $contextStorage;
+        $this->metricFactory = $metricFactory ?? new StreamFactory();
+        $this->resource = $resource;
+        $this->attributesFactory = $attributesFactory;
+        $this->clock = $clock;
+        $this->instrumentationScopeFactory = $instrumentationScopeFactory;
+        $this->metricReader = $metricReader;
+        $this->stalenessHandlerFactory = $stalenessHandlerFactory;
+        $this->criteriaViewRegistry = new CriteriaViewRegistry();
+        $this->viewRegistry = new FallbackViewRegistry($this->criteriaViewRegistry, [
```

Comment:
Not valid anymore, has to be configured by metric reader - [View](https://github.com/open-telemetry/opentelemetry-specification/blob/793354957a11b839d677084d074a1ee11f474b73/specification/metrics/sdk.md#view):
> The aggregation (optional) to be used. If not provided, the SDK MUST apply a default aggregation configurable on the basis of instrument kind according to the MetricReader instance.